packages/backend/src/open_payments/payment/outgoing/service.ts

```diff
@@ -272,6 +279,16 @@ async function validateGrant(
     return false
   }
 
+  //lock grant
+  //TODO: update to use objection once it supports forNoKeyUpdate
+  await deps
+    .knex<GrantModel>('grants')
```

Comment:
I'm assuming this is using the transaction that is `deps.knex`?

Reply:
That is how I understood it, yes.